crates/op-rbuilder/src/generator.rs

```diff
@@ -114,6 +122,26 @@ where
         &self,
         attributes: <Builder as PayloadBuilder<Pool, Client>>::Attributes,
     ) -> Result<Self::Job, PayloadBuilderError> {
+        let cancel_token = if self.ensure_only_one_payload {
+            // Cancel any existing payload first
+            {
+                let last_payload = self.last_payload.lock().unwrap();
```

Comment:
I wonder why you lock twice here? If you think that token can hold the lock its still not going to help since you hold it while cancelling